Compare serverless and EC hosted

leemthompo opened the pull request:
## 🔭 👁️ [URL preview](https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/894/deploy-manage/deploy/elastic-cloud/differences-from-other-elasticsearch-offerings)

- Edits the existing "[Serverless differences](https://docs-v3-preview.elastic.dev/elastic/docs-content/tree/main/deploy-manage/deploy/elastic-cloud/differences-from-other-elasticsearch-offerings)" file which previously only contained Elasticsearch information
  - Moves it up to **Elastic Cloud** level and tries to incorporate as much info as I could currently put together across the various project types
  
  
### Some notes:

- We don't have links to platform-level features, which I think is fine at least in a first iteration, or could add those in a follow-up if we deem that important.
- I deliberately chose not to link to features that are discontinued/deprecated in 9.0.
- There's some platform-level features sprinkled into the solutions, which we might wanna pull up into the platform table, like API-keys, RBAC etc. but again probably OK right now and they don't have links, just like other platform-level features. They are an artefact of how folks filled out the spreadsheet a little non-uniformly.

florent-leborgne commented:
Nice it's much more complete.

I'd put the architectural differences first as it's the very first decision point.

There's also a table in the "Elastic Cloud" parent page that does a sort of high level comparison, not sure we want to keep it, maybe instead turn in into a link to this updated page?

leemthompo commented:
latest commit:


* Restructured and expanded feature comparison tables for better readability
* Alphabetized feature entries in all tables
* Added new features to comparison tables (e.g., "BYO-Key for Encryption at Rest", "Private Link", "Shard management")
* Updated status of several features from "WIP" to "Planned" with roadmap links
* Clarified deprecation status of certain features (e.g., "discontinued in 9.0") for ent search stuff

leemthompo commented:
thanks @lucabelluccini! I'm sure you'll want to tweak/correct my additions

mdbirnstiehl reviewed (approved):
Looks good on the features for Observability. One thing I noticed is different language, "Not available in Serverless", "Not supported in Serverless architecture", "Not available in Security projects", "Not supported in Serverless". If these all mean the same thing, we might want to align the language.

nastasha-solomon commented on deploy-manage/deploy/elastic-cloud/differences-from-other-elasticsearch-offerings.md:
```diff
@@ -1,81 +1,176 @@
+| **Kibana Alerts** | ✅ | ✅ | |
+| **Kibana navigation** | Standard layout | Different layout | UI differences in Security projects |
+| **LogsDB** | Optional | ✅ | - Enabled by default <br>- Cannot be disabled |
+| **Observability Logs UI** | ✅ | ❌ | Not available in Security projects |
```
This can be removed.

```suggestion
```

nastasha-solomon commented on deploy-manage/deploy/elastic-cloud/differences-from-other-elasticsearch-offerings.md:
```diff
@@ -1,81 +1,176 @@
+| **Observability Logs UI** | ✅ | ❌ | Not available in Security projects |
+| **Role-based access control** | ✅ | Limited | Core RBAC functionality supported - [Learn more](https://www.elastic.co/guide/en/serverless/current/security-rbac.html) |
+| **SIEM capabilities** | ✅ | ✅ | Core functionality supported |
+| **Watcher** | ✅ | ❌ | Use **Kibana Alerts** instead |
```
This can be removed as well.
```suggestion
```

nastasha-solomon commented on deploy-manage/deploy/elastic-cloud/differences-from-other-elasticsearch-offerings.md:
```diff
@@ -1,81 +1,173 @@
+| **API keys** | ✅ | ✅ | |
+| **Defend for Containers integration** | ✅ (deprecated in 9.0) | ❌ | Not supported in Serverless architecture |
+| **Endpoint security** | ✅ | ✅ | |
+| **Field and document level security** | ✅ | ✅ | |
```
Sorry one more suggestion - I _think_ you can remove this too. Elastic Security features mainly focus on monitoring and securing your environment. This entry seems more related to securing your data within Elasticsearch. 

```suggestion
```
